packages/core/lib/commands/migrate/run.js

```diff
@@ -1,15 +1,29 @@
 module.exports = async function (options) {
   const WorkflowCompile = require("@truffle/workflow-compile").default;
   const { Environment } = require("@truffle/environment");
-  const Config = require("@truffle/config");
   const determineDryRunSettings = require("./determineDryRunSettings");
   const prepareConfigForRealMigrations = require("./prepareConfigForRealMigrations");
   const runMigrations = require("./runMigrations");
   const setUpDryRunEnvironmentThenRunMigrations = require("./setUpDryRunEnvironmentThenRunMigrations");
+  const loadConfig = require("../../loadConfig");
+  const OS = require("os");
+  const TruffleError = require("@truffle/error");
   const tmp = require("tmp");
   tmp.setGracefulCleanup();
 
-  const config = Config.detect(options);
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#migrate" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
```

Comment:
I think Truffle sets`options.network = develop` when using the `truffle develop` REPL console that would trigger this error condition if the user enters `migrate --url`. We could either error better here, **or** filter the command in the console logic. What do you think?

Reply:
When we are already inside the REPL, then it ignores the `--url` option and does the migrations on the `develop` network (managed ganache with http://127.0.0.1:9545). It doesn't trigger the error condition. I think its better to error out the cause for that to let the users know that this option is not applicable inside the REPL.

Reply:
Please ignore my previous comment @cds-amal. I wasn't testing this with the correct branch. Your interpretation is correct that inside the develop REPL , `migrate --url` will trigger the above mentioned error. We can error better here to display something like "`migrate --url` is not supported here".

Reply:
Yea, I prefer error better at the REPL input processing seam.